docs/source/trainer.rst

```diff
@@ -157,6 +157,28 @@ Once you're done training, feel free to run the test set!
 
     trainer.test(test_dataloaders=test_dataloader)
 
+If you need to save the predictions, nothing simpler ! This will works with ``DDP``
+
+.. code-block:: python
+
+    class MyModel(LightningModule):
+
+        def test_step(self, batch, batch_idx):
+
+            preds = ...
+
+            # save predictions
+            # we need a unique int id to identify each sample.
+            predictions = []
+            for idx, id in enumerate(batch["id"]):
+                predictions.append({"id": id, "preds": preds[idx]})
+
+            self.trainer.predictions.add(predictions)
```

Comment:
can we make this one self.add_predictions(predictions) or something similar? users shouldn't have to manually call the trainer for something that essential.

Reply:
Good idea !

Reply:
could we add an argument that will auto id the predictions?
Then if pass in a tensor then it will just give each prediction the next available id:
``self.add_predictions(prediction, auto_id=True)``

Reply:
Removed the trainer entirely.

Reply:
@SkafteNicki isn't that what batch_idx is for?

Reply:
batch_idx tracks the batch_id. Here we are tracking the sample.
Furthermore, when running with DDP, batch_idx starts from 0 to dataset_size // num_world_size, whereas each process will get different data.

Reply:
Hey @SkafteNicki, `self.add_predictions(prediction, auto_id=True)` could be added in the future, but it will be slightly harder, as we would have to get the sample id if the user performs a shuffle.